### PR TITLE
checker: check nested struct field with required attr (fix #10913)

### LINKED ISSUE #10913: nested auto initialized struct with required attribute doesn't give an error
**OS:** WSL2 Ubuntu 20.04

**What did you do?**
```v
struct Foo {
	bar Bar
}
struct Bar {
	egg Egg
}

struct Egg {
	name string [required]
}

fn main() {
	f := Foo{}
	println(f)
}
```

**What did you expect to see?**
Foo contains a value that needs manual initialization (error)
 
**What did you see instead?**
it compiles and prints
```
Foo{
    bar: Bar{
        egg: Egg{
            name: ''
        }
    }
}
```

similar problem described here: https://github.com/vlang/v/issues/10823

### LINKED ISSUE #16400: `[required]` in nested field init does not work when default value.
**V version: 0.3.2 12f0131** 
**OS: ArchLinux** 

**What did you do?**

```v
struct Foo{
	n int [required]
}

struct Bar{
	foo Foo
}

fn main(){
	b := Bar{}
	println(b)
}
```
**What did you expect to see?**
Error or warn that `Bar.foo.n` must be initialized.
 
**What did you see instead?**
No error.
```
> v run .
Bar{
    foo: Foo{
        n: 0
    }
}
```

### LINKED ISSUE #16423: `[noinit]` in struct's field does not work as default value
**V version: 0.3.2 1a4d1ae** 
**OS: ArchLinux** 

 
Some specifications have not been determined as to what is the correct behavior, and this would be one of them.
We should make a list of such issues that are left in limbo because the specifications cannot be determined.

**What did you do?**

```
.
├── bug.v
├── mod
│   └── mod.v
└── v.mod
```
bug.v
```v
import mod

struct Bar{
	foo mod.Foo
}

fn main(){
	b := Bar{}
	println(b)
	f := mod.Foo{}
}
```
mod.v
```v
module mod

[noinit]
pub struct Foo{
pub:
	n int
}
```

**What did you expect to see?**
`b := Bar{}` should be an error also.
 
**What did you see instead?**
```
> v run bug.v
bug.v:10:2: warning: unused variable: `f`
    8 |     b := Bar{}
    9 |     println(b)
   10 |     f := mod.Foo{}
      |     ^
   11 | }
bug.v:10:11: error: struct `mod.Foo` is declared with a `[noinit]` attribute, so it cannot be initialized with `mod.Foo{}`
    8 |     b := Bar{}
    9 |     println(b)
   10 |     f := mod.Foo{}
      |              ~~~~~
   11 | }
```

### PR DESCRIPTION
**Pull request by yuyi98**
This PR check nested struct field with required attr (fix #10913, fix #16400, fix #16423).

- Check nested struct field with required attr.
- Add test.

```v
struct Foo {
	bar Bar
}
struct Bar {
	egg Egg
}

struct Egg {
	name string [required]
}

fn main() {
	f := Foo{}
	println(f)
}

PS D:\Test\v\tt1> v run .
tt1.v:13:7: error: field `Egg.name` must be initialized
   11 | 
   12 | fn main() {
   13 |     f := Foo{}
      |          ~~~~~
   14 |     println(f)
   15 | }
```

```v
import mod

struct Bar{
	foo mod.Foo
}

fn main(){
	b := Bar{}
	println(b)
}

PS D:\Test\v\tt1> v run .
tt1.v:8:7: error: struct `mod.Foo` is declared with a `[noinit]` attribute, so it cannot be initialized with `mod.Foo{}`
    6 |
    7 | fn main(){
    8 |     b := Bar{}
      |          ~~~~~
    9 |     println(b)
   10 | }
```

### CODE REVIEW

File: vlib/net/http/header.v [hunk 1]
```diff
@@ -6,7 +6,6 @@ module http
 import strings
 
 // Header represents the key-value pairs in an HTTP header
-[noinit]
```
Comment (spytheman) [inline vlib/net/http/header.v:9]: Why was that removed?

It seems unrelated to the change at first glance 🤔 .

**Review by spytheman — APPROVED** [review]

File: vlib/net/http/header.v (hunk 1 above)
Comment (yuyi98) [reply vlib/net/http/header.v:9]: `net.http.Header` used in many structs, e.g. pub struct websocket.Client. 
It cannot init if using default struct init.

Comment (spytheman) [reply vlib/net/http/header.v:9]: Indeed ... I saw that by trying to compile `examples/web_crawler/web_crawler.v` with V from this PR. I'll have to think more about it, since I am not sure why it was `[noinit]` in the first place 🤔 .
